src/api/apiClient.ts

```diff
@@ -0,0 +1,128 @@
+// reference: https://github.com/16Yongjin/tutoring-app/tree/main/src/api
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import { getCookie } from 'utils/ts/cookie';
+// eslint-disable-next-line import/no-cycle
+import { APIRequest } from './interfaces/APIRequest';
+import { APIResponse } from './interfaces/APIResponse';
+import { APIError } from './interfaces/APIError';
+
+const API_URL = process.env.REACT_APP_API_PATH;
+
+export const HTTPMethod = {
+  GET: 'GET',
+  POST: 'POST',
+  PUT: 'PUT',
+  DELETE: 'DELETE',
+} as const;
+
+export type UnionHttpMethod<T> = T[keyof T];
```

Comment:
```suggestion
export const HTTP_METHOD = {};
export type HTTPMethod = typeof HTTP_METHOD[keyof typeof HTTP_METHOD];
```
다음과 같이 바꿔주세요. 불필요한 제네릭 사용입니다.

Reply:
제네릭 제거와 해당 관련 변수 수정하였습니다
[0251961](https://github.com/BCSDLab/KOIN_WEB_RECODE/pull/7/commits/025196143f1455e8e757c00eba11cc1d72357576)